### PR TITLE
Events CRUD and interface

### PR DESCRIPTION
**Pull request by gtalha07**
PR Description:
- Expose Calendar Events CRUD APIs.
- Implement Calendar Events interface in spaces and pages.
- Implement CRUD sheets.
- Add calendar events providers and routing.

Bugs and Blockers: 
- Calendar events not updating after editing.
- ```setLink``` API is missing on ```CalendarEventDraft``` rust model.
- Invites and joining event features (future milestones).

Preview Video:

https://github.com/acterglobal/a3/assets/68579938/03c3c744-1311-44aa-9412-086171a22a55

### CODE REVIEW

File: app/lib/features/events/providers/events_provider.dart [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  late Stream<void> _listener;
+  Future<List<CalendarEvent>> _getEvents() async {
+    return (await arg.calendarEvents()).toList(); // this might throw internally
+  }
+
+  @override
+  Future<List<CalendarEvent>> build(Space arg) async {
+    final client = ref.watch(clientProvider)!;
+    final spaceId = arg.getRoomId();
+    _listener = client.subscribe('$spaceId::CALENDAR'); // stay up to date
```
Comment (gnunicorn) [inline app/lib/features/events/providers/events_provider.dart:19]: aren't we doing lowercase keys? [Yeah, we are](https://github.com/acterglobal/a3/blob/main/native/core/src/statics.rs#L11).

```suggestion
    _listener = client.subscribe('$spaceId::calendar'); // stay up to date
```

File: app/lib/features/events/widgets/events_calendar.dart [hunk 2]
```diff
@@ -78,7 +39,7 @@ class EventsCalendar extends ConsumerWidget {
                       style: Theme.of(context).textTheme.bodyLarge,
                     ),
                     subtitle: Text(
-                      formatDt(e),
+                      formatDt(e) + '\t UTC (GMT+0)',
```
Comment (gnunicorn) [inline app/lib/features/events/widgets/events_calendar.dart:42]: shouldn't we format in the local timezone?

**Review by gnunicorn — APPROVED** [review]
just a minor thing, looks good at a glance overall.

File: app/lib/common/utils/routes.dart [hunk 3]
```diff
@@ -34,12 +34,18 @@ enum Routes {
...
   // -- pins
 
   pins('/pins'),
   pin('/pins/:pinId'),
 
+  // -- events
+  createEvent('/events/create'),
+  calendarEvent('/events/:calendarId'),
+  editCalendarEvent('/events/:calendarId/edit'),
+
```
[the remark this answers is not in the document]
Comment (gtalha07) [reply app/lib/common/utils/routes.dart:48]: You mean ```/events``` page outside from space shell? Yeah it isn't there yet. But we do have space events page

File: app/lib/features/events/providers/events_provider.dart (hunk 1 above)
Comment (gtalha07) [reply app/lib/features/events/providers/events_provider.dart:19]: Sure, will replace it.

File: app/lib/features/events/widgets/events_calendar.dart (hunk 2 above)
Comment (gtalha07) [reply app/lib/features/events/widgets/events_calendar.dart:42]: Should we ? That's possible but designs preview UTC date. I guess I could modify this utility.

**Review by gnunicorn — APPROVED** [review]

**Pull request merged**